9.tcga-classify/scripts/nbconverted/interpret-all-feature-ensemble-model.py

```diff
@@ -16,128 +16,165 @@
+# In[8]:
+
+
+all_scores_df = all_scores_df.assign(big_feature_id=all_scores_df.algorithm + "_" +
+                                     all_scores_df.feature.astype(str) + "_" +
+                                     all_scores_df.seed.astype(str) + "_" +
+                                     all_scores_df.z.astype(str) + "_signal")
+all_scores_df = all_scores_df.assign(abs_z_score=all_scores_df.z_score.abs())
+
+
+# In[9]:
+
+
+all_coef_scores_df = (
+    full_coef_id_df
+    .merge(all_scores_df,
            how='left',
-           left_on=['full_feature', 'algorithm', 'seed'],
-           right_on=['use_feature', 'feature_alg', 'seed'])
-    .drop(['model_type', 'feature_x', 'feature_y', 'signal'], axis='columns')
+           left_on="feature",
+           right_on="big_feature_id")
     .sort_values(by=['abs', 'abs_z_score'], ascending=False)
     .reset_index(drop=True)
 )
 
-top_biobombe_df = (
-    biobombe_df
-    .groupby('full_feature')
+all_coef_scores_df.head()
+
+
+# In[10]:
+
+
+# Explore the biobombe scores for specific DAE features
+top_n_features = 5
+
+biobombe_df = (
+    all_coef_scores_df
+    .groupby('big_feature_id')
     .apply(func=lambda x: x.abs_z_score.nlargest(top_n_features))
     .reset_index()
-    .merge(biobombe_df
+    .merge(all_coef_scores_df
            .reset_index(),
-           right_on=['index', 'abs_z_score', 'full_feature'],
-           left_on=['level_1', 'abs_z_score', 'full_feature'])
-    .drop(['level_1', 'index'], axis='columns')
-    .sort_values(by=['weight', 'z_score'], ascending=False)
+           right_on=['index', 'abs_z_score', 'big_feature_id'],
+           left_on=['level_1', 'abs_z_score', 'big_feature_id'])
+    .drop(['level_1', 'index', 'feature_x',
+           'algorithm_x', 'seed_x',
+           'model_type', 'algorithm_y',
+           'feature_y', 'seed_y', 'z'], axis='columns')
+    .sort_values(by=['abs', 'abs_z_score'], ascending=False)
+    .reset_index(drop=True)
 )
-    
-    
-print(top_biobombe_df.shape)
-top_biobombe_df
 
+print(biobombe_df.shape)
+biobombe_df.head(20)
 
-# In[7]:
+
+# In[11]:
+
+
+# Output biobombe scores applied to the all feature ensemble model
+file = os.path.join('results', 'tcga_tp53_classify_top_biobombe_scores_all_feature_ensemble_model_table.tsv')
+biobombe_df.to_csv(file, sep='\t', index=False)
+
+
+# ## Detect the highest contributing variables
+
+# In[12]:
+
+
+neg_biobombe_df = biobombe_df.query("weight < 0")
+pos_biobombe_df = biobombe_df.query("weight > 0")
+
+top_neg_variables_df = neg_biobombe_df.groupby("variable")['weight'].sum().sort_values(ascending=True)
+top_pos_variables_df = pos_biobombe_df.groupby("variable")['weight'].sum().sort_values(ascending=False)
+
+
+# In[13]:
+
+
+full_result_df = pd.DataFrame(pd.concat([top_pos_variables_df, top_neg_variables_df]))
+full_result_df = (
+    full_result_df
+    .assign(abs_weight=full_result_df.weight.abs())
+    .sort_values(by='abs_weight', ascending=False)
+)
+
+full_result_df.head()
```

Comment:
Why do you have the raw weight and the absolute value in the final table?

Reply:
Here I care about contribution, positive and negative. I sort by the absolute value weight so that negative weights can also be scanned in the top of the table